tests/e2e.rs

```diff
@@ -1396,6 +1397,112 @@ async fn test_e2e_sso_linking() {
 	);
 }
 
```

Comment:
We should add tests for the edge cases I mention further up; IDs that happen to be valid base64/hex, but aren't intended to be treated as such originally.